src/renderer/components/Editor/SquareBracket.tsx

```diff
@@ -0,0 +1,45 @@
+import { Box } from '@mui/system';
+import { useState } from 'react';
+import colors from 'renderer/colors';
+import SquareBracketHover from './SquareBracketHover';
+
+interface Props {
+  isLast: boolean;
+  isTakeGroupOpened: boolean;
+}
+
+const SquareBracket = ({ isLast, isTakeGroupOpened }: Props) => {
+  const bottomWidth = isLast || !isTakeGroupOpened ? '2px' : '0px';
+
+  const [isHovering, setIsHovering] = useState(false);
```

Comment:
`isHovering` seems like it's only being used to adjust the opacity of the children. Instead, you could define this in CSS. Something along the lines of (syntax might not be correct but the idea is there)

```
&:hover {
  & .innerBracket {
    opacity: 0.5;  
  }
}
```

Which would work if you used `styled()` to define the "squareBracket" box as we have for other components, and give the children a classname

Reply:
I really liked your solution and I gave this a shot. The class names have to be unique so that it could only turn on the hover of SquareBracketHover component. If all the class names are the same then all the components will have the hover effect when you hover over 1.

I tried using a template literal so that I can have unique class names per SquareBracket component I got a syntax error.

Here is a screenshot: 
![image](https://user-images.githubusercontent.com/86769131/190383527-38abade7-a185-446b-99df-6ae4abeda1f1.png)


Im going to stash these changes and i'll show you them tomorrow

Reply:
Very cool, you're so close 😅 

Because it's an object, if the key is a variable and not a string literal then you need to put it in square brackets. I'll give you a more generic example and you can apply to your specific situation

Example:

```
const keyName = 'myKey';
const object = {
    [keyName]: 'someValue'
}
```

The above will produce this object: `{ "myKey": "someValue" }`
Whereas without the square brackets you get: `{ "keyName": "someValue" }`

Reply:
The javascript portion of that worked, 

![image](https://user-images.githubusercontent.com/86769131/190523932-85876f59-53b1-4870-aab7-58699c96c870.png)


But it is not hovering, when i use the dev tools it shows nothing up. 

![image](https://user-images.githubusercontent.com/86769131/190524008-7195cfcb-7ea4-4bf0-b598-14501d292645.png)


I pushed those changes on another branch, I made a PR so you can see the diff: https://github.com/chloebrett/mlvet/pull/279. Let me know if you can spot whats wrong.